clang/test/3C/bodiless.c

```diff
@@ -0,0 +1,60 @@
+// RUN: 3c %s > %S/temp_bodiless.c
+// RUN: %clang -c %S/temp_bodiless.c
+// RUN: FileCheck -match-full-lines --input-file %S/temp_bodiless.c %s
+// RUN: rm %S/temp_bodiless.c
+
+/***********************************************/
+/* Tests that functions without bodies         */
+/* are marked wild when there is no interface  */
+/***********************************************/
+
+static int *foo1(void) { return (void *)0; }
+void test1() {
+  int *a = foo1();
+}
+//CHECK _Ptr<int> a = foo1();
+
+static int *foo2(void);
```

Comment:
What happens if you add a test for `static int *foo2a(void) { }` ? See comment on the code, above.

Reply:
Isn't this the same as test1? a body exists, so it's treated as normal without being constrained to wild as if it were external. There are no unsafe uses, so it is re-written as a checked pointer. The lack of return is an problem, but I think that's beyond the scope of this issue. Any non-void function missing a return should be marked as unsafe, though CheckedC will catch that error.

Reply:
No, because the body of `foo2a` is empty but it nevertheless declares that it returns something. We want to make sure that the `int *` returned stays as an `int *`.

Reply:
I'm sorry, my reading of the issue was that it specifically dealt with undefined static functions, so that's what this pull request solves. Is the scope intended to be defined functions that have no statements, or more generally following control flow to find branches without return statements?

Reply:
See my comment above: I am asking when `G->hasBody()` returns false. Does it just return false when you have something like `static void foo(void);` or does it also return false when you have `static void foo(void) { }` ? I am wondering whether these two are deemed equivalent by clang. If so, then you want to have a test like this. If not, no need, I agree.

Reply:
Sorry for the delay. From testing, `hasBody()` means a function has a definition, not just a declaration. So, for `static void foo(void) { }` `hasBody()` returns true.

Let me know what the next steps are for this issue.

Reply:
OK, let's drop it (i.e., stick with what you have). Thanks for checking.